pkg/kube/activity.go

```diff
@@ -171,23 +188,36 @@ func (k *PipelineActivityKey) GetOrCreate(activities typev1.PipelineActivityInte
 		},
 		Spec: v1.PipelineActivitySpec{},
 	}
-	if activities == nil {
+	activitiesClient := jxClient.JenkinsV1().PipelineActivities(ns)
+
+	if activitiesClient == nil {
 		log.Warn("Warning: no PipelineActivities client available!")
 		return defaultActivity, create, nil
 	}
-	a, err := activities.Get(name, metav1.GetOptions{})
+	a, err := activitiesClient.Get(name, metav1.GetOptions{})
 	if err != nil {
 		create = true
 		a = defaultActivity
 	}
 	oldSpec := a.Spec
+	oldLabels := a.Labels
+
+	// lazy create missing sourcerepository entries
+	if a.Labels[v1.LabelSourceRepository] == "" {
```

Comment:
🐛 If there are no labels applied, then `a.Labels` will be nil, meaning that this will throw a nil reference exception (I'm about 99% sure of this, but _may_ be wrong)